st2common/st2common/jinja/filters/complex_type.py

```diff
@@ -0,0 +1,32 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+__all__ = [
+    'to_complex'
+]
+
+
+def to_complex(value):
+    result = json.dumps(value)
+
+    if isinstance(result, unicode):
+        try:
+            result = str(result)
+        except UnicodeEncodeError:
+            pass
```

Comment:
IIRC, we allow unicode in other places (I did a bug fix for allowing unicode quite a long time ago) so we should probably also allow unicode here.